applications/GeoMechanicsApplication/tests/cpp_tests/test_time_stepping.cpp

```diff
@@ -0,0 +1,154 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "testing/testing.h"
+#include "custom_workflows/time_step_executor.h"
+#include "custom_workflows/strategy_wrapper.hpp"
+#include "solving_strategies/strategies/solving_strategy.h"
+
+using namespace Kratos;
+
+
+namespace
+{
+
+class ProcessSpy : public Process
+{
+public:
+    void ExecuteInitializeSolutionStep() override {mSolutionStepInitializedCalls += 1;}
+    void ExecuteFinalizeSolutionStep()   override {mSolutionStepFinalizedCalls   += 1;}
+
+    unsigned int NumberOfExecuteInitializeSolutionStepCalls() const {return mSolutionStepInitializedCalls;}
+    unsigned int NumberOfExecuteFinalizeSolutionStepCalls()   const {return mSolutionStepFinalizedCalls;}
+
+private:
+    unsigned int mSolutionStepInitializedCalls = 0;
+    unsigned int mSolutionStepFinalizedCalls   = 0;
+};
+
+class DummyStrategyWrapper : public StrategyWrapper
+{
+public:
+    explicit DummyStrategyWrapper::DummyStrategyWrapper(bool convergence_needed) :
+        mConvergenceNeeded ( convergence_needed ? TimeStepEndState::ConvergenceState::converged : TimeStepEndState::ConvergenceState::non_converged) {}
+    [[nodiscard]] TimeStepEndState::ConvergenceState GetConvergenceState(const TimeStepEndState& rEndState) override { return mConvergenceNeeded;};
+    [[nodiscard]] std::size_t                        GetNumberOfIterations()                                const override { return 4;};
+    [[nodiscard]] double                             GetEndTime()                                           const override { return 10.;};
+    void Initialize()             override {mSolverStrategyInitializeCalls += 1;}
+    void InitializeSolutionStep() override {mSolverStrategyInitializeSolutionStepCalls += 1;}
+    void Predict()                override {mSolverStrategyPredictCalls += 1;}
+    bool SolveSolutionStep()      override {mSolverStrategySolveSolutionsStepCalls += 1;
+                                            return true;}
+    void FinalizeSolutionStep()   override {mSolverStrategyFinalizeSolutionStepCalls += 1;}
```

Comment:
Also here I would prefer to use the pre-increment operator.

Reply:
done